atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -697,6 +699,35 @@ private void joinWithRetry(
         });
   }
 
+  public CompletableFuture<Void> leave() {
+    final CompletableFuture<Void> future = new CompletableFuture<>();
+
+    threadContext.execute(
+        () -> {
+          final var leaving = cluster.getLocalMember();
+          final var receiver =
+              membershipService.getMembers().stream()
+                  .filter(member -> !member.id().equals(leaving.memberId()))
+                  .findAny()
+                  .orElseThrow()
+                  .id();
+          protocol
+              .leave(receiver, LeaveRequest.builder().withLeavingMember(leaving).build())
+              .whenCompleteAsync(
+                  (response, error) -> {
+                    if (error != null) {
+                      future.completeExceptionally(error);
+                    } else if (response.status() == Status.OK) {
+                      future.complete(null);
```

Comment:
:question: Here the partition does not transition to inactive, until we explicitly shut it down. Is that right?